core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -873,6 +875,39 @@ object Chunk {
       case _: BitChunk       => ClassTag.Boolean.asInstanceOf[ClassTag[A]]
     }
 
+  /**
+   * The maximum number of elements in the buffer for fast append.
+   */
+  private val BufferSize: Int =
+    100
+
+  private final case class Add[A](start: Chunk[A], buffer: Array[AnyRef], bufferUsed: Int, chain: AtomicInteger)
```

Comment:
Maybe good to call this `AppendN`, to pave the way for `PrependN` (Vector `+:`). Then we can support efficient prepends and appends.

Reply:
Yes.